src/main/java/org/folio/tlib/postgres/TenantPgPool.java

```diff
@@ -5,11 +5,21 @@
 import io.vertx.pgclient.PgConnectOptions;
 import io.vertx.pgclient.PgPool;
 import java.util.List;
+import javax.validation.constraints.NotNull;
 import org.folio.tlib.postgres.impl.TenantPgPoolImpl;
 
 public interface TenantPgPool extends PgPool {
 
-  static TenantPgPool pool(Vertx vertx, String tenant) {
+  /**
+   * create tenant pool for tenant.
+   * @param vertx vert.x instance
+   * @param tenant tenant name.
+   * @return pool.
+   */
+  static TenantPgPool pool(Vertx vertx, @NotNull String tenant) {
+    if (tenant == null) {
+      throw new IllegalArgumentException("Tenant must not be null");
```

Comment:
This looks redundant given that we use the NotNull annotation above

Reply:
It doesn't kick in

Reply:
The `tenant == null` check should be moved into `TenantPgPoolImpl.tenantPgPool`.